python/lsst/obs/base/butler_tests.py

```diff
@@ -143,7 +143,11 @@ def _test_exposure(self, name):
         self.assertEqual(exp.getDetector().getId(), self.butler_get_data.detectorIds[name])
         self.assertEqual(exp.getDetector().getName(), self.butler_get_data.detector_names[name])
         self.assertEqual(exp.getDetector().getSerial(), self.butler_get_data.detector_serials[name])
-        self.assertEqual(exp.getFilter().getName(), self.butler_get_data.filters[name])
+        # obs_test does not have physical filters, so include a fallback
+        exposureFilter = exp.getFilterLabel()
+        filterName = exposureFilter.physicalLabel if exposureFilter.hasPhysicalLabel() \
+            else exposureFilter.bandLabel
```

Comment:
I'm torn on whether this should be an `x=if y else z` or `if y:\nx=...\nelse:\n`. Since it's long enough to need a new line anyway, I don't know that this is an improvement. I never quite know myself what to do in this kind of case.

Reply:
I prefer conditional expressions when I want to make it clear that _something_ is getting assigned, and it's just a question of what. This is especially true in languages that can't guarantee that a variable is set exactly once.